drivers/hmis/app/graphql/types/hmis_schema/service_type.rb

```diff
@@ -25,10 +25,12 @@ class HmisSchema::ServiceType < Types::BaseObject
     field :hud_record_type, HmisSchema::Enums::Hud::RecordType, null: true
     field :hud_type_provided, HmisSchema::Enums::ServiceTypeProvided, null: true
     field :category, String, null: false
+    field :category_record, HmisSchema::ServiceCategory, null: false
     field :form_definitions, [Forms::FormDefinition], null: false, description: 'Definitions that are specified for this service type'
 
     # object is a Hmis::Hud::CustomServiceType
 
+    # TODO(#5737) - remove this in favor of category_record, then rename category_record to category ?
```

Comment:
Let's discuss - this is to add support for matching the service category in the frontend to a pick list option which is keyed by category ID https://github.com/greenriver/hmis-frontend/pull/969/files#diff-817dbd1069fa5529fc67ea3079fdcb4a54eee9430fb15dc14fa9b279dc8d5b63R82.

Reply:
maybe it would be less disruptive to just add a field `field :category_id, ID, null: false`, instead of resolving a new object which duplicates the category name?

Another option would be to have the picklist just be names (not IDs), though it looks like the database doesn't validate the uniqueness of category names, so we would need the in order to confidently use names only.

Reply:
Yes, the first option is what I had implemented initially to minimize disruption. However, I thought that since this will need to be a 2-release feature for backwards compat anyway (search TODO(#5737) for other follow-up changes), I thought the disruption isn't so bad and it may be worth it to better match our usual pattern.

However, if you are also worried about another kind of disruption (likelihood of introducing mistakes/bugs?) I'm happy to update it to take that approach.

Reply:
I'm not worried about bugs as much as the effort involved to make these changes. If I understand correctly the change would need to span 3 releases:
* add `categoryRecord`, start fetching it from the frontend  (this release)
* change existing `category` type to be of type `HmisSchema::ServiceCategory` and update frontend to fetch _that_ instead; deprecate `categoryRecord` (next release)
* remove deprecated `categoryRecord` (next-next release)

Spanning 3 releases for a naming change just feels like a lot of effort relative to the benefit. If we remove the `"then rename category_record to category"` step, then its only these 2 releases, which seems OK since you already have other API deprecations to clean up. That said I have no preference for `categoryRecord { name, id }` vs  `category, categoryId`, so would be totally fine with leaving as-proposed. (With the suggestion that we don't re-name it again later).

Reply:
Yes, good point that 3 releases seems a bit much. I'm proposing to rename it to `serviceCategory` and then keep that name in the long term, avoiding the 3rd release. I guess it's not perfect, but it does seem to match the pattern in a couple places elsewhere (like `projectType` for example seems similar-spirited to me?)

Reply:
that works, thanks!